### PR TITLE
Modified report endpoint to accept only requests with x-www-form-urlencoded form data

### LINKED ISSUE #293: Modify report endpoints to only accept forms and not json data?
Perhaps maintenance and usage would be easier. Right now if you go to swagger you don't see the params that you could pass to the endpoints any more..

### PR DESCRIPTION
**Pull request by northwestwitch**
### Added/Changed/Fixed
- Closes #293

image

### How to test
- [x] Deploy on stage and test creating the report with a scout sample

### Expected outcome
- Report should show up

## Review
- [x] Tests executed by CR
- [x] "Merge and deploy" approved by SA

### This [version](https://semver.org/) is a
- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: src/chanjo2/endpoints/report.py [hunk 1]
```diff
@@ -52,18 +54,23 @@ async def demo_report(request: Request, db: Session = Depends(get_session)):
 @router.post("/report", response_class=HTMLResponse)
 async def report(
     request: Request,
+    build=Annotated[Builds, Form(...)],
```
Comment (northwestwitch) [inline src/chanjo2/endpoints/report.py:57]: Followed the FastAPI tutorial ---> https://fastapi.tiangolo.com/tutorial/request-forms/

**Comment by northwestwitch** [comment]
> > Hi @seallard, I've changed that endpoint we discussed last week. You could review if you have some time perhaps? Thanks!
> 
> Sure! To understand the context better, where is the endpoint used? Is it only by the scout web app?

Yes, from here: https://github.com/Clinical-Genomics/scout/blob/4dd74e1057ccc541ca120ff26c59f01effc82e97/scout/server/blueprints/cases/templates/cases/chanjo2_form.html#L32

**Comment by northwestwitch** [comment]
Also from the report page itself actually, to update the report:

image

File: src/chanjo2/endpoints/report.py [hunk 2]
```diff
@@ -52,18 +54,23 @@ async def demo_report(request: Request, db: Session = Depends(get_session)):
...
+    samples=Annotated[str, Form(...)],
+    interval_type=Annotated[IntervalType, Form(...)],
+    completeness_thresholds=Annotated[Optional[str], Form(None)],
+    ensembl_gene_ids=Annotated[Optional[str], Form(None)],
+    hgnc_gene_ids=Annotated[Optional[str], Form(None)],
+    hgnc_gene_symbols=Annotated[Optional[str], Form(None)],
+    case_display_name=Annotated[Optional[str], Form(None)],
+    panel_name=Annotated[Optional[str], Form("Custom panel")],
+    default_level=Annotated[Optional[int], Form(DEFAULT_COVERAGE_LEVEL)],
```
Comment (seallard) [inline src/chanjo2/endpoints/report.py:66]: Only the request and db parameters are used here and the others can be removed

File: src/chanjo2/endpoints/report.py [hunk 3]
```diff
@@ -52,18 +54,23 @@ async def demo_report(request: Request, db: Session = Depends(get_session)):
...
 
     start_time = time.time()
-    request_headers: str = request.headers.get("Content-Type")
-
     try:
-        if request_headers == "application/json":
-            report_query = ReportQuery(**await request.json())
-        else:
-            report_query = ReportQuery.as_form(await request.form())
+        report_query = ReportQuery.as_form(await request.form())
```
[the remark this answers is not in the document]
Comment (seallard) [reply src/chanjo2/endpoints/report.py:73]: Did the json content format not work? If it is just about making the swagger docs report the params, I think you can keep it. It is a nice-to-have and the consumer of the endpoint can specify which format of the content

File: src/chanjo2/endpoints/report.py [hunk 4]
```diff
@@ -52,18 +54,23 @@ async def demo_report(request: Request, db: Session = Depends(get_session)):
 @router.post("/report", response_class=HTMLResponse)
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply src/chanjo2/endpoints/report.py:54]: Yes, very long lists of genes, that exceed the size of a get request params

**Comment by northwestwitch** [comment]
> A bit unclear why the endpoint uses post when it retrieves data. I'm guessing the query params are too big for a get?

Exactly!

> A bit unclear why the endpoint uses post when it retrieves data. I'm guessing the query params are too big for a get?

Exactly. Thanks!

**Comment by northwestwitch** [comment]
> I don't really see what is gained by removing support for the json content type, but I'm not familiar with this web API or usage.

We have another endpoint that accepts json. Having this one accepting only forms would simplify things, I think

**Pull request merged**